_posts/2017-05-15-rust-at-two-years.md

```diff
@@ -0,0 +1,356 @@
+---
+layout: post
+title: "Two years of Rust"
+author: Carol (Nichols || Goulding)
+description: "Rust, two years after 1.0"
+---
+
+It's been two years since Rust 1.0 was released. Happy second birthday, Rust!
+
+![Group picture from RustFest Berlin][group-pic]{:class="center"}
+
+*Rustaceans at RustFest Berlin, September 2016. Picture by Fiona Castiñeira*
+
+[group-pic]: /images/2017-05-Second-Birthday/rustfest-berlin.jpeg
+
+Rust, a systems programming language focused on safety, speed, and concurrency,
+has now achieved stability without stagnation by maintaining backwards
+compatibility with version 1.0 for two years while also making many
+improvements. Conveniently, Rust's birthday is a bit under halfway through
+2017, which makes this a great time to reflect not only on the progress in the
+last year but also on the progress of our [2017 Roadmap] goals.
+
+[2017 Roadmap]: https://blog.rust-lang.org/2017/02/06/roadmap.html
+
+After reading this post, if you'd like to give us your feedback on how we're
+doing and where Rust should focus next, please fill out our [2017 State of Rust
+survey].
+
+[2017 State of Rust survey]: https://blog.rust-lang.org/2017/05/03/survey.html
+
+But first, let's do the numbers!
+
+## Rust in numbers
+
+A lot has happened since [Rust's first birthday]:
+
+<!--
+Will rerun these just before publishing:
+
+commits:
+`git log --since=2016-05-15 --until=2017-05-15 --oneline | wc`
+
+contributors:
+`git log --since=2016-05-15 --until=2017-05-15 --oneline --format="%aN" | sort | uniq | wc`
+
+rfcs:
+`git ls-tree a4a22d7c5dd71724bb2cd0fe2db5026338d0b270 text/ | wc` => there were 255 RFCs a year ago
+`git ls-tree HEAD text/ | wc` => there are 310 RFCs now
+310 - 255 =  55
+
+crates:
+`select count(*) from crates where date(created_at) >= '2016-05-15';`
+
+friends:
+https://github.com/rust-lang/rust-www/compare/bb7f26b...master#diff-ee76f5da00d603d5e34d053a3dc63395
+-->
+
+- 10,583 [commits] by 655 contributors added to the core repository;
+- 55 [RFCs] merged;
+- 9 major releases and 2 patch releases shipped;
+- 4,303 new [crates] published;
+- 284 library stabilizations;
+- 10 languages [rust-lang.org] has been translated into;
+- 48 new companies [running Rust in production][friends];
+- 2 new teams (Docs and Style);
```

Comment:
Also, Infra (which will be fully announced any moment now) and the Unsafe Guidelines strike team.